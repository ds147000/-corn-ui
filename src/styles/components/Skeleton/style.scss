.xrk-skeleton {
  width: 100%;
  background-color: #fff;
  position: relative;
  overflow: hidden;

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 400px;
    height: 100%;
    z-index: 1;
    background: linear-gradient(to right , rgba(0 , 0 , 0 , 0) , rgba(255 , 255 , 255 , 0.6) , rgba(0 , 0 , 0 , 0));
    animation: xrkSkeletonMove 1.5s infinite;
  }

  &-item {
    background-color: #ebeef6;
  }
}

@keyframes xrkSkeletonMove {
  0% {
    transform: translate(-50vw , 0);
  }

  35% {
    transform: translate(-50vw , 0);
  }

  100% {
    transform: translate(100vw , 0);
  }
}
