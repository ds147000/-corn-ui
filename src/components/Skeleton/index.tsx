import { View } from '@tarojs/components'
import React from 'react'
import { SkeletonItem, SkeletonItemProps } from './item'

export interface SkeletonProps {
  children: React.ReactElement<SkeletonItemProps>
}

type SkeletonComponents = React.FC & { Item: typeof SkeletonItem }

const Skeleton: SkeletonComponents = ({ children }) => {
  return (
    <View className="xrk-skeleton" >
      {children}
    </View>
  )
}

Skeleton.Item = SkeletonItem

export default Skeleton
