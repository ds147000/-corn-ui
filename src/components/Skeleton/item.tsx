/* eslint-disable no-magic-numbers */
import React from 'react'
import { View } from '@tarojs/components'
import { transformRem } from '../../utils'

export interface SkeletonItemProps {
  /** 宽度，自动转换合适750单位。 默认 690 */
  width?: number
  /** 高度，自动转换合适750单位。 默认 46 */
  height?: number
  /** 顶部外边度，自动转换合适750单位。 默认 15 */
  top?: number
  /** 底部外边度，自动转换合适750单位。 默认 15 */
  bottom?: number
  /** 左部外边度，自动转换合适750单位。 默认 30 */
  left?: number
  /** 右部外边度，自动转换合适750单位。 默认 30 */
  right?: number
}

export const SkeletonItem: React.FC<SkeletonItemProps> = ({
  width = 690,
  height = 46,
  top = 15,
  bottom = 15,
  left = 30,
  right = 30
}) => {
  const _width = transformRem(width)
  const _height = transformRem(height)
  const _top = transformRem(top)
  const _left = transformRem(left)
  const _bottom = transformRem(bottom)
  const _right = transformRem(right)
  const _borderRadius = height > 46 ? transformRem(15) : transformRem(8)

  return (
    <View style={{ paddingTop: _top, paddingBottom: _bottom, paddingLeft: _left, paddingRight: _right }} >
      <View className="xrk-skeleton-item" style={{ width: _width, height: _height, borderRadius: _borderRadius }} />
    </View>
  )
}
