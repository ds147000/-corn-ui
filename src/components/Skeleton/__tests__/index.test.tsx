import React from 'react'
import { render } from '@testing-library/react'
import Skeleton from '../index'

describe('Skeleton', () => {
  test('快照', () => {
    const screen = render(
      <Skeleton>
        <Skeleton.Item height={345} width={345} top={7.5} bottom={7.5} left={15} right={15} />
        <Skeleton.Item height={58} top={7.5} bottom={7.5} left={15} right={15} />
        <Skeleton.Item height={23} top={7.5} bottom={7.5} left={15} right={15} />
        <Skeleton.Item />
      </Skeleton>
    )

    expect(screen.container).toMatchSnapshot()
  })
})
