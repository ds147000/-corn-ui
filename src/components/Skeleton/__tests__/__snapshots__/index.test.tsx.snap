// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton 快照 1`] = `
<div>
  <div
    class="xrk-skeleton"
  >
    <div
      class=""
      style="padding: 0.1600rem 0.3200rem 0.1600rem 0.3200rem;"
    >
      <div
        class="xrk-skeleton-item"
        style="width: 7.3600rem; height: 7.3600rem; border-radius: 0.3200rem;"
      />
    </div>
    <div
      class=""
      style="padding: 0.1600rem 0.3200rem 0.1600rem 0.3200rem;"
    >
      <div
        class="xrk-skeleton-item"
        style="width: 14.7200rem; height: 1.2373rem; border-radius: 0.3200rem;"
      />
    </div>
    <div
      class=""
      style="padding: 0.1600rem 0.3200rem 0.1600rem 0.3200rem;"
    >
      <div
        class="xrk-skeleton-item"
        style="width: 14.7200rem; height: 0.4907rem; border-radius: 0.1707rem;"
      />
    </div>
    <div
      class=""
      style="padding: 0.3200rem 0.6400rem 0.3200rem 0.6400rem;"
    >
      <div
        class="xrk-skeleton-item"
        style="width: 14.7200rem; height: 0.9813rem; border-radius: 0.1707rem;"
      />
    </div>
  </div>
</div>
`;
